Modules/posixmodule.c

```diff
@@ -1391,15 +1390,16 @@ convertenviron(void)
     d = PyDict_New();
     if (d == NULL)
         return NULL;
-#if defined(WITH_NEXT_FRAMEWORK) || (defined(__APPLE__) && defined(Py_ENABLE_SHARED))
-    if (environ == NULL)
-        environ = *_NSGetEnviron();
-#endif
 #ifdef MS_WINDOWS
     /* _wenviron must be initialized in this way if the program is started
        through main() instead of wmain(). */
     _wgetenv(L"");
     e = _wenviron;
+#elif defined(WITH_NEXT_FRAMEWORK) || (defined(__APPLE__) && defined(Py_ENABLE_SHARED))
```

Comment:
See my previous comment. This block can be removed when using the \#define I mention there.

Reply:
I prefer to see an explicit _NSGetEnviron() rather than macros black magic.

Reply:
Neither option is obviously better. Just calling _NSGetEnviron does make it more explicit what's going on.